sklearn/preprocessing/data.py

```diff
@@ -1674,22 +1674,22 @@ def _transform_selected(X, transform, selected="all", copy=True):
         return transform(X)
     else:
         X_sel = transform(X[:, ind[sel]])
-        X_not_sel = X[:, ind[not_sel]]
+        X_not_sel = X[:, ind[not_sel]].astype(dtype)
```

Comment:
Scipy sparse matrices until recently did not support boolean fancy indexing. But that's why we have `sklearn.utils.safe_mask`.